database/database.go

```diff
@@ -0,0 +1,809 @@
+			panic("Violated: if min_period changed, timestamp is erased: !dbc.InTest || err != nil || t.mustGetCh(channel.Descriptor_) == nil || t.mustGetCh(channel.Descriptor_).MinPeriod == channel.MinPeriod || t.mustGetTs(channel.Descriptor_) == nil")
+		default:
+			// Pass
+		}
+	}()
+
+	var serialized []byte
+	serialized, err = proto.Marshal(channel)
+	if err != nil {
+		err = fmt.Errorf("failed to serialize the channel: "+
+			"%s", err.Error())
+		return
+	}
+
+	encodedKey := Descriptor(channel.Descriptor_).Encode()
+
+	prevChan, err := t.GetChannel(channel.Descriptor_)
+	if err != nil {
+		err = fmt.Errorf("failed to fetch the channel: %s", err.Error())
+		return
+	}
+
+	err = t.lmdbTxn.Put(t.channelDbi, encodedKey, serialized, 0)
+	if err != nil {
+		err = fmt.Errorf("failed to put the channel: %s", err.Error())
+		return
+	}
+
+	if prevChan != nil && prevChan.MinPeriod != channel.MinPeriod {
+		var ts *Timestamp
+		ts, err = t.GetTimestamp(channel.Descriptor_)
+		if err != nil {
+			err = fmt.Errorf("failed to fetch the timestamp: %s", err.Error())
+			return
+		}
+		if ts != nil {
+			err = t.removeTimestamp(channel.Descriptor_)
+			if err != nil {
+				err = fmt.Errorf("failed to erase the timestamp: %s", err.Error())
+				return
+			}
+		}
+	}
+
+	return
+}
+
+// PutTimestamp inserts a Timestamp in the database, keyed on its descriptor.
+//
+// PutTimestamp requires:
+// * !dbc.InTest || t.access == RelayAccess
+// * Timestamp != nil
+// * *Timestamp > 0
+//
+// PutTimestamp ensures:
+// * !dbc.InTest || err != nil || t.mustGetTs(string(descriptor)) != nil
+// * !dbc.InTest || err != nil || t.mustCountTs() > 0
+func (t *Txn) PutTimestamp(descriptor Descriptor,
+	Timestamp *Timestamp) (err error) {
+	// Pre-conditions
+	switch {
+	case !(!dbc.InTest || t.access == RelayAccess):
+		panic("Violated: !dbc.InTest || t.access == RelayAccess")
+	case !(Timestamp != nil):
+		panic("Violated: Timestamp != nil")
+	case !(*Timestamp > 0):
+		panic("Violated: *Timestamp > 0")
+	default:
+		// Pass
+	}
+
+	// Post-conditions
+	defer func() {
+		switch {
+		case !(!dbc.InTest || err != nil || t.mustGetTs(string(descriptor)) != nil):
+			panic("Violated: !dbc.InTest || err != nil || t.mustGetTs(string(descriptor)) != nil")
+		case !(!dbc.InTest || err != nil || t.mustCountTs() > 0):
+			panic("Violated: !dbc.InTest || err != nil || t.mustCountTs() > 0")
+		default:
+			// Pass
+		}
+	}()
+
+	encodedTs := Timestamp.Encode()
+	encodedKey := descriptor.Encode()
+
+	err = t.lmdbTxn.Put(t.timestampDbi, encodedKey, encodedTs, 0)
+	if err != nil {
+		err = fmt.Errorf("failed to put the Timestamp: %s", err.Error())
+		return
+	}
+
+	return
+}
+
+// RemoveChannel removes a channel from the database.
+//
+// RemoveChannel requires:
+// * !dbc.InTest || t.mustGetCh(descriptor) != nil
```

Comment:
Fair point. I'll change it.

Reply:
P.S.: I'll also remove the 404 error code in the server and substitute with a 204. We want to avoid the client program to take incorrect actions based on the assumption the DELETE failed (404 is an error coce, after all).